nfa_to_dfa/Graph.java

```diff
@@ -1,99 +1,118 @@
 package nfa_to_dfa;
 
+import java.util.HashMap;
 import java.util.ArrayList;
 
 /**
- * While intended to be a parent class for the DFA class, the NFA class, and others, 
- * the Graph class can be utilized in many applications. Utilizes {@link Node}, {@link GraphEdge}
+ * While intended to be a parent class for the DFA class, the NFA class, and
+ * others,
+ * the Graph class can be utilized in many applications. Utilizes {@link Node},
+ * {@link GraphEdge}
  * 
  * @see GraphEdge
  * @see Node
  * @see DFA
  * @see NFA
- * @author Ethan Leyden 
+ * @author Ethan Leyden
  */
 public class Graph {
-    private ArrayList<ArrayList<GraphEdge>> adj; //adjacency matrix with edges
-    private ArrayList<Node> nodes; //node labels, where their index represents the index in the adj matrix
+    private ArrayList<ArrayList<GraphEdge>> adj; // adjacency matrix with edges
+    private ArrayList<Node> nodes; // node labels, where their index represents the index in the adj matrix
     private int num_nodes = 0;
-    private HashMap<String, Integer> ids; //Look up a label, get an id
-    private HashMap<Integer, Integer> idx; //Look up an id, get an index in the adjacency matrix. 
+    private HashMap<String, Integer> ids; // Look up a label, get an id
+    private HashMap<Integer, Integer> idx; // Look up an id, get an index in the adjacency matrix.
+
     /**
      * Creates a new Graph object with no nodes or (obviously) edges between nodes.
      */
     public Graph() {
         this.adj = new ArrayList<>();
         this.nodes = new ArrayList<>();
-        labels = new HashMap();
-        ids = new HashMap(); 
+        HashMap<String, Integer> labels = new HashMap();
```

Comment:
This should just be `labels = new HashMap();` as declaring the type makes it local